diff --git a/myocamlbuild.ml b/myocamlbuild.ml
index 12ee8b5..d0fc279 100644
--- a/myocamlbuild.ml
+++ b/myocamlbuild.ml
@@ -893,36 +893,3 @@ let () =
 
          | _ ->
              ())
-
-(* Compile the wiki version of the Ocamldoc.
-
-   Thanks to Till Varoquaux on usenet:
-   http://www.digipedia.pl/usenet/thread/14273/231/
-
-*)
-
-let ocamldoc_wiki tags deps docout docdir =
-  let tags = tags -- "extension:html" in
-  Ocamlbuild_pack.Ocaml_tools.ocamldoc_l_dir tags deps docout docdir
-
-let () =
-  try
-    let wikidoc_dir =
-      let base = Ocamlbuild_pack.My_unix.run_and_read "ocamlfind query wikidoc" in
-      String.sub base 0 (String.length base - 1)
-    in
-
-    Ocamlbuild_pack.Rule.rule
-      "ocamldoc: document ocaml project odocl & *odoc -> wikidocdir"
-      ~insert:`top
-      ~prod:"%.wikidocdir/index.wiki"
-      ~stamp:"%.wikidocdir/wiki.stamp"
-      ~dep:"%.odocl"
-      (Ocamlbuild_pack.Ocaml_tools.document_ocaml_project
-         ~ocamldoc:ocamldoc_wiki
-         "%.odocl" "%.wikidocdir/index.wiki" "%.wikidocdir");
-
-    tag_file "lwt-api.wikidocdir/index.wiki" ["apiref";"wikidoc"];
-    flag ["wikidoc"] & S[A"-i";A wikidoc_dir;A"-g";A"odoc_wiki.cma"]
-
-  with Failure e -> () (* Silently fail if the package wikidoc isn't available *)
